File: app/src/main/java/com/ojnexus/core/model/JudgeId.kt
package com.ojnexus.core.model

/**
 * Identifier for an online judge. Display names are fixed proper nouns (not translatable UI
 * copy) and rendered in the app's uppercase telemetry style.
 *
 * Adapters own all judge-specific behavior; core code only ever refers to judges through this
 * enum so a broken adapter can never leak into another judge's data path.
 */
enum class JudgeId(
    val id: String,
    val displayName: String,
) {
    CODEFORCES("codeforces", "CODEFORCES"),
    ATCODER("atcoder", "ATCODER"),
    LUOGU("luogu", "LUOGU"),
    ;

    companion object {
        fun fromId(id: String): JudgeId? = entries.firstOrNull { it.id == id }
    }
}

File: windows/src/OjNexus.Windows.Core/Domain/JudgeId.cs
namespace OjNexus.Windows.Core.Domain;

public enum JudgeId
{
    Codeforces,
    AtCoder,
    Luogu,
}

public static class JudgeIdParser
{
    public static bool TryParse(string value, out JudgeId judge)
    {
        if (string.Equals(value, nameof(JudgeId.Codeforces), StringComparison.OrdinalIgnoreCase))
        {
            judge = JudgeId.Codeforces;
            return true;
        }

        if (string.Equals(value, nameof(JudgeId.AtCoder), StringComparison.OrdinalIgnoreCase))
        {
            judge = JudgeId.AtCoder;
            return true;
        }

        if (string.Equals(value, nameof(JudgeId.Luogu), StringComparison.OrdinalIgnoreCase))
        {
            judge = JudgeId.Luogu;
            return true;
        }

        judge = default;
        return false;
    }
}
